test/unit/app/jobs/dynamic_tool_destination/mockGalaxy.py

```diff
@@ -43,7 +43,7 @@ def __init__(self, file_name, file_ext, value):
     def get_metadata(self):
         return self.metadata
 
-    def get_file_name(self, **_kwargs):
+    def get_file_name(self, _sync_cache=True):
```

Comment:
This is minor. I assume the underscore in the parameter name is meant to indicate that it's never used (but must be present for correct method overriding)? If so, I think it might be better to keep the naming consistent with the real Dataset: that way it's clear it is *not* different or special in any way, so it won't require any further thought, and if in doubt, a simple grep reveals it's not used. Besides, if it *is* used at some point, we won't have to rename it back. (same goes for the same param in the tool_util and model packages). Unless, of course, there's other reasoning involved here.

Reply:
No reason, except that I thought Mypy could complain about unused parameters. But I did not check if it does, so I can change that (it is not the only place, so I can change that everywhere).

Actually, originally I wanted to have **kwargs in all unimportant declarations, like Mock stuff etc, but then Mypy did complain.

Then another thought was to use **kwargs everywhere in `get_file_name` which would make it more flexible (I'll be adding another parameter in the next commit), but also less explicit, so I did not do that

Reply:
If you don't mind, let's not use `**kwargs**` here. Understanding and debugging code becomes more complicated down the road: with a sufficient amount of indirection (which is typical..), sometimes it becomes hard to determine what exactly might be passed. In my opinion, using `**kwargs**` is indeed helpful when there's a function which takes a variable number of parameters by design (i.e., is expected to be called with different sets of arguments), there may be more than a few, and functionally they have a similar purpose - so grouping them would make sense). But here, I think, explicitly listing the parameters might be more appropriate. And yes, as far as I know, Mypy won't complain about unused parameters.